lib/src/cache/reference_cache.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import 'package:w_common/src/cache/cache.dart';
+
+/// Maintains the number of references for an instance of an cache value.
+class ReferenceCache<TIdentifier, TValue> extends Cache<TIdentifier, TValue> {
```

Comment:
I am totally up for a better name. I would like to solicit more feedback and get a list of other suggestions before changing the code.